kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -443,8 +443,50 @@ private Optional<Connection> getAppliedConnection(Device dev) throws DBusExcepti
                     .of(this.dbusConnection.getRemoteObject(NM_BUS_NAME, connectionPath.getPath(), Connection.class));
         } catch (DBusExecutionException e) {
             logger.debug("Could not find applied connection for {}, caused by", dev.getObjectPath(), e);
-            return Optional.empty();
         }
+
+        Optional<Connection> connectionToReturn = Optional.empty();
+        
+        try {
+            logger.info("Active connection not found, looking for avaliable connections.");
+
+            Settings settings = this.dbusConnection.getRemoteObject(NM_BUS_NAME, NM_SETTINGS_BUS_PATH,
+                    Settings.class);
+            
+            List<DBusPath> connectionPath = settings.ListConnections();
+            
+            Properties deviceProperties = this.dbusConnection.getRemoteObject(NM_BUS_NAME, dev.getObjectPath(), Properties.class);
+            String interfaceName = deviceProperties.Get(NM_DEVICE_BUS_NAME, NM_DEVICE_PROPERTY_INTERFACE); 
+            String targetConnectionName = String.format("kura-%s-connection", interfaceName);
+
+            boolean isFirst = true;
+            for (DBusPath path : connectionPath) {
+
+                Connection workingConnection = this.dbusConnection.getRemoteObject(NM_BUS_NAME, path.getPath(), Connection.class);
+                
+                Map<String, Map<String, Variant<?>>> workingSettings = workingConnection.GetSettings();
+                
+                String workingConnectionName = (String) workingSettings.get("connection").get("id").getValue();
+                String workingConnectionId = (String) workingSettings.get("connection").get("uuid").getValue();
+                
+                if (workingConnectionName.contains(targetConnectionName)) {
```

Comment:
We're looking for an exact match, not partial

```suggestion
                if (workingConnectionName.equals(targetConnectionName)) {
```

Reply:
Will do! 

My thought process was that we discussed potentially adding an extra identifier to the name.